brig/cmd/brig/commands/project_create.go

```diff
@@ -0,0 +1,540 @@
+
+	"github.com/Azure/brigade/pkg/brigade"
+	"github.com/Azure/brigade/pkg/storage/kube"
+)
+
+const projectCreateUsage = `Create a new project.
+
+Create a new project by answering questions or supplying a configuration file.
+
+Project names are typically in the form 'ORG/PROJECT' or 'USER/PROJECT". For
+example, Azure/brigade or deis/empty-testbed.
+
+A Brigade project provides a context in which a brigade.js file is executed.
+Projects are frequently tied to (Git) source code repositories, and contain
+configuration data such as secrets, Kubernetes-specific directives, and
+settings for gateways or workers.
+
+If a file is provided (-f), then the project defaults will be set from the file.
+
+The user will be prompted to answer questions to build the configuration. If -f
+is specified in conjunction with -x/--no-prompts, then the values in the file will
+be used without prompting the user for any changes.
+`
+
+var (
+	projectCreateConfig      string
+	projectCreateOut         string
+	projectCreateFromProject string
+	projectCreateDryRun      bool
+	projectCreateNoPrompts   bool
+	projectCreateReplace     bool
+)
+
+var defaultProject = &brigade.Project{
+	Name: "deis/empty-testbed",
+	Repo: brigade.Repo{
+		Name:     "github.com/deis/empty-testbed",
+		CloneURL: "https://github.com/deis/empty-testbed.git",
+		SSHKey:   "",
+	},
+	Secrets:      map[string]string{},
+	SharedSecret: "",
+	Github:       brigade.Github{},
+	Kubernetes:   brigade.Kubernetes{},
+	Worker: brigade.WorkerConfig{
+		PullPolicy: "IfNotPresent",
+	},
+}
+
+func init() {
+	project.AddCommand(projectCreate)
+	flags := projectCreate.Flags()
+	flags.StringVarP(&projectCreateConfig, "config", "f", "", "Path to JSON Kubernetes Secret.")
+	flags.StringVarP(&projectCreateFromProject, "from-project", "p", "", "Retrieve the given project from Kubernetes and use it to set the default values.")
+	flags.StringVarP(&projectCreateOut, "out", "o", "", "File where configuration should be saved. The configuration is stored as a JSON Kubernetes Secret")
+	flags.BoolVarP(&projectCreateDryRun, "dry-run", "D", false, "Do not send the config to Kubernetes")
+	flags.BoolVarP(&projectCreateNoPrompts, "no-prompts", "x", false, "Do not prompt the user for input. Use this with -f to skip prompts.")
+	flags.BoolVarP(&projectCreateReplace, "replace", "r", false, "Replace an existing project with this revised one. Use this to modify existing projects.")
+}
+
+var projectCreate = &cobra.Command{
+	Use:   "create",
+	Short: "create a new Brigade project",
+	Long:  projectCreateUsage,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if projectCreateFromProject != "" && projectCreateConfig != "" {
+			return errors.New("cannot specify both -f/--config and -p/--from-project")
+		}
+		return createProject(cmd.OutOrStderr())
+	},
+}
+
+func createProject(out io.Writer) error {
+	c, err := kubeClient()
+	if err != nil {
+		return err
+	}
+
+	store := kube.New(c, globalNamespace)
```

Comment:
Configuring `globalNamespace` which is `default` means the namespace configured through the survey is not used - so all projects are created in the `default` namespace, regardless of the value passed.